vulture_os/authentication/openid/views.py

```diff
@@ -103,13 +103,16 @@ def edit(request, object_id=None, api=False):
 
 
 def test_provider(request):
-    form = OpenIDRepositoryForm(request.POST)
+    provider = OpenIDRepository.objects.filter(provider=request.POST.get('provider'), client_id=request.POST.get('client_id'), client_secret=request.POST.get('client_secret')).first()
```

Comment:
if a user tries to test an existing Connector with new parameters, those changed values won't be taken into account.

Instead, a good way of testing would be to implement a utility function taking the necessary parameters, and then use it both here (with the values taken directly from the POST values) and in the current `retrieve_config()` function.

This way, there would be no need to instantiate any Form or Model in the `test_provider()` function